go/vt/vttablet/common/config.go

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vttablet
+
+import (
+	"encoding/json"
+	"fmt"
+	"maps"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+type VReplicationConfig struct {
+	ExperimentalFlags       int64
+	NetReadTimeout          int
+	NetWriteTimeout         int
+	CopyPhaseDuration       time.Duration
+	RetryDelay              time.Duration
+	MaxTimeToRetryError     time.Duration
+	RelayLogMaxSize         int
+	RelayLogMaxItems        int
+	ReplicaLagTolerance     time.Duration
+	HeartbeatUpdateInterval int
+	StoreCompressedGTID     bool
+	ParallelInsertWorkers   int
+	TabletTypesStr          string
+
+	VStreamPacketSize                      int
+	VStreamPacketSizeOverride              bool
+	VStreamDynamicPacketSize               bool
+	VStreamDynamicPacketSizeOverride       bool
+	VStreamBinlogRotationThreshold         int64
+	VStreamBinlogRotationThresholdOverride bool
+
+	Overrides map[string]string
+}
+
+var configMutex sync.Mutex
+var DefaultVReplicationConfig *VReplicationConfig
```

Comment:
Do we really want to use a pointer? The behavior feels undefined when anyone can change the defaults. Is that OK or even desirable? I feel like the default should be set once via sync.Once and then the Gets return a CloneVT of it. I think that simplifies the logical and better defines the behavior while simplifying the logic, lessening locking, and likely faster (so everything can be on the stack). Maybe not though...

This file could use some comments so that it's more clear what the desired behaviors are and aren't.

Reply:
We temporarily modify some of the flags in unit tests and having that pointer was the simplest way of doing it. I can change it all to use Getters/Setters and make that private if required: see one usage herehttps://github.com/vitessio/vitess/pull/16583/files#diff-77c3f361c9a41c782f5cd18667f3d1ce25d1fc9415fe28c80412e4f5b42694bcL63-L69 

Will add more comments to that file.